docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,106 @@
+
+Mozilla generally favors a rapid release model to long cycle times.  Firefox desktop adopted rapid releases over a decade ago and there is current work to move both the mobile applications and our services towards rapid releases.  Although there are some costs to rapid releases, as a company we've decided that we're willing to accept them.
+
+Application services should consider moving towards a rapid release model.  The same reasoning that led other teams to adopt rapid releases also applies to us.  Furthermore, the mobile teams are moving towards rapid releases which may depend on us also adopting them.  Finally, we are in the process of moving towards a single monorepo for android code, with potential to add iOS code and/or move into the moz-central monorepo. This would require that application services code is taken by all other consumer applications in the monorepo as soon as it's merged.  Reducing the amount of time between our releases and when applications take those releases brings us closer to that monorepo world.
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process (Jan 2023 - Feb 2023)
+        * Update our branches to match how firefox-android works: a `main` branch that is used for the current nightly builds and a `releases_v[N]` for each previous release.
+        * Automate a nightly release of our `main` branch.
+          * Create a commit that updates our version number to something like `[firefox-version].0-nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * Note: we will no longer be using semver-style versions and also not following semver semantics.  We will be releasing breaking changes in a nightly cycle without bumping the minor version.
+          * Create a GitHub release from the branch with a tag based on the version number / date (`v[firefox-version].0-nightly-[YYYYMMDD]`). This will cause maven packages to be published, start the `rust-components-swift` CI, etc.
+        * Manually handle releases from the `release_v*` branches whenever new code is committed to those branches
```

Comment:
I'm a bit confused on what this means -- specifically, "whenever new code is committed". Does this essentially mean we cut when we feel a release is ready? 

Since we will have nighties, feels like we only need these releases to coincide with when the consumers need a stable release (which would probably be at the end of the nightly cycles?).

Reply:
I was thinking more about patch releases here.  The "whenever new code is committed" text was trying to say that we should make a patch release right after the merge.  There's no point in having unreleased code sit around in these branches.  Hopefully the next text is more clear about this.